packages/eslint-plugin/src/rules/prefer-string-starts-ends-with.ts

```diff
@@ -0,0 +1,647 @@
+      node: TSESTree.Node,
+    ): { isStartsWith: boolean; isEndsWith: boolean; text: string } | null {
+      const evaluated = getStaticValue(node, globalScope);
+      if (evaluated == null || !(evaluated.value instanceof RegExp)) {
+        return null;
+      }
+
+      const { source, flags } = evaluated.value;
+      const isStartsWith = source.startsWith('^');
+      const isEndsWith = source.endsWith('$');
+      if (
+        isStartsWith === isEndsWith ||
+        flags.includes('i') ||
+        flags.includes('m')
+      ) {
+        return null;
+      }
+
+      const text = parseRegExpText(source, flags.includes('u'));
+      if (text == null) {
+        return null;
+      }
+
+      return { isEndsWith, isStartsWith, text };
+    }
+
+    /**
+     * Fix code with using the right operand as the search string.
+     * For example: `foo.slice(0, 3) === 'bar'` → `foo.startsWith('bar')`
+     * @param fixer The rule fixer.
+     * @param node The node which was reported.
+     * @param kind The kind of the report.
+     * @param negative The flag to fix to negative condition.
+     */
+    function* fixWithRightOperand(
+      fixer: RuleFixer,
+      node: TSESTree.BinaryExpression,
+      kind: 'start' | 'end',
+      negative: boolean,
+    ): IterableIterator<RuleFix> {
+      // left is CallExpression or MemberExpression.
+      const leftNode = (node.left.type === 'CallExpression'
+        ? node.left.callee
+        : node.left) as TSESTree.MemberExpression;
+      const propertyRange = getPropertyRange(leftNode);
+
+      if (negative) {
+        yield fixer.insertTextBefore(node, '!');
+      }
+      yield fixer.replaceTextRange(
+        [propertyRange[0], node.right.range[0]],
+        `.${kind}sWith(`,
+      );
+      yield fixer.replaceTextRange([node.right.range[1], node.range[1]], ')');
+    }
+
+    /**
+     * Fix code with using the first argument as the search string.
+     * For example: `foo.indexOf('bar') === 0` → `foo.startsWith('bar')`
+     * @param fixer The rule fixer.
+     * @param node The node which was reported.
+     * @param kind The kind of the report.
+     * @param negative The flag to fix to negative condition.
+     */
+    function* fixWithArgument(
+      fixer: RuleFixer,
+      node: TSESTree.BinaryExpression,
+      kind: 'start' | 'end',
+      negative: boolean,
+    ): IterableIterator<RuleFix> {
+      const callNode = node.left as TSESTree.CallExpression;
+      const calleeNode = callNode.callee as TSESTree.MemberExpression;
+
+      if (negative) {
+        yield fixer.insertTextBefore(node, '!');
+      }
+      yield fixer.replaceTextRange(
+        getPropertyRange(calleeNode),
+        `.${kind}sWith`,
+      );
+      yield fixer.removeRange([callNode.range[1], node.range[1]]);
+    }
+
+    return {
+      // foo[0] === "a"
+      // foo.charAt(0) === "a"
+      // foo[foo.length - 1] === "a"
+      // foo.charAt(foo.length - 1) === "a"
+      // @ts-ignore
```

Comment:
Instead of using `@ts-ignore`, couldn't you could silence this by explicitly calling `.join(',')` on the array?